src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -60,19 +61,45 @@ public FunctionIndexer(ITriggerBindingProvider triggerBindingProvider, IBindingP
                 throw new ArgumentNullException("singletonManager");
             }
 
+            if (trace == null)
+            {
+                throw new ArgumentNullException("trace");
+            }
+
             _triggerBindingProvider = triggerBindingProvider;
             _bindingProvider = bindingProvider;
             _activator = activator;
             _executor = executor;
             _singletonManager = singletonManager;
             _jobAttributeAssemblies = GetJobAttributeAssemblies(extensions);
+            _trace = trace;
         }
 
         public async Task IndexTypeAsync(Type type, IFunctionIndexCollector index, CancellationToken cancellationToken)
         {
             foreach (MethodInfo method in type.GetMethods(PublicMethodFlags).Where(IsJobMethod))
             {
-                await IndexMethodAsync(method, index, cancellationToken);
+                try
+                {
+                    await IndexMethodAsync(method, index, cancellationToken);
+                }
+                catch (FunctionIndexingException fex)
+                {
+                    // Route the indexing exception through the TraceWriter pipeline
+                    _trace.Error(fex.Message, fex);
+
+                    // If the error has been marked as handled, ignore the indexing exception
+                    // and continue on to the rest of the methods. In this case the method in
+                    // error simply won't be running in the JobHost.
+                    if (fex.Handled)
+                    {
+                        continue;
```

Comment:
This goes beyond the added trace support no?

So at the time exception is thrown, it may already be marked handled? I guess I don't have context to understand what that means.